x/cardchain/keeper/vote_right.go

```diff
@@ -19,39 +20,41 @@ func (k Keeper) GetVoteReward(ctx sdk.Context) sdk.Coin {
 	return reward
 }
 
-// AddVoteRight Adds a voteright for a certain card to a certain user
-func (k Keeper) AddVoteRight(ctx sdk.Context, userAddress sdk.AccAddress, cardId uint64) error {
-	user, err := k.GetUser(ctx, userAddress)
-	if err != nil {
-		return err
-	}
-
-	gottenCard := k.Cards.Get(ctx, cardId)
-	if !gottenCard.BalanceAnchor {
-		right := types.NewVoteRight(cardId, ctx.BlockHeight()+k.GetParams(ctx).VotingRightsExpirationTime)
-		user.VoteRights = append(user.VoteRights, &right)
-		k.SetUser(ctx, userAddress, user)
+// AddVoteRightToUser Adds a voteright for a certain card to a certain user
+func (k Keeper) AddVoteRightToUser(ctx sdk.Context, user *types.User, cardId uint64) {
+	card := k.Cards.Get(ctx, cardId)
+	if !card.BalanceAnchor {
+		if !slices.Contains(user.VotableCards, cardId) && !slices.Contains(user.VotedCards, cardId) {
```

Comment:
Look good. 1 Person 1 Vote is MSG of democracy

Reply:
Very goood, MSG of codereview.